physics/ksp_system_test.cpp

```diff
@@ -0,0 +1,469 @@
+                 EvaluatePosition(t, /*hint=*/nullptr);
+    };
+    auto const degrees_of_freedom = [t, &ephemeris](
+        KSPCelestial const& celestial) {
+      return ephemeris->trajectory(celestial.body)->
+                 EvaluateDegreesOfFreedom(t, /*hint=*/nullptr);
+    };
+    auto const jool_position = position(jool);
+
+    for (auto const* moon : moons) {
+      Length const separation = (jool_position - position(*moon)).Norm();
+      Sign separation_change = Sign(separation - last_separations[moon]);
+      if (separation_change != last_separation_changes.at(moon)) {
+        extremal_separations_in_m[moon].emplace_back(last_separations[moon] /
+                                                     Metre);
+        times_in_s[moon].emplace_back((t - 1 * Hour - ksp_epoch) / Second);
+      }
+      last_separations[moon] = separation;
+      last_separation_changes.at(moon) = separation_change;
+    }
+
+    tylo_bop_separations_in_m.emplace_back(
+        (position(tylo) - position(bop)).Norm() / Metre);
+    pol_bop_separations_in_m.emplace_back(
+        (position(pol) - position(bop)).Norm() / Metre);
+
+    {
+      // KSP's osculating elements.
+      auto const bop_elements =
+          KeplerOrbit<KSP>(*jool.body,
+                           MasslessBody(),
+                           degrees_of_freedom(bop) - degrees_of_freedom(jool),
+                           t).elements_at_epoch();
+      bop_eccentricities.emplace_back(bop_elements.eccentricity);
+      bop_inclinations_in_degrees.emplace_back(bop_elements.inclination /
+                                               Degree);
+      bop_nodes_in_degrees.emplace_back(
+          bop_elements.longitude_of_ascending_node / Degree);
+      bop_arguments_of_periapsis_in_degrees.emplace_back(
+          bop_elements.argument_of_periapsis / Degree);
+    }
+
+    {
+      BarycentreCalculator<DegreesOfFreedom<KSP>, GravitationalParameter>
+          innermost_jool_system;
+      for (auto const* celestial : {&jool, &laythe, &vall, &tylo}) {
+        innermost_jool_system.Add(degrees_of_freedom(*celestial),
+                                  celestial->body->gravitational_parameter());
+      }
+      auto const bop_jacobi_elements =
+          KeplerOrbit<KSP>(
+              MassiveBody(innermost_jool_system.weight()),
+              *bop.body,
+              degrees_of_freedom(bop) - innermost_jool_system.Get(),
+              t).elements_at_epoch();
+      bop_jacobi_eccentricities.emplace_back(bop_jacobi_elements.eccentricity);
+      bop_jacobi_inclinations_in_degrees.emplace_back(
+          bop_jacobi_elements.inclination / Degree);
+      bop_jacobi_nodes_in_degrees.emplace_back(
+          bop_jacobi_elements.longitude_of_ascending_node / Degree);
+      bop_jacobi_arguments_of_periapsis_in_degrees.emplace_back(
+          bop_jacobi_elements.argument_of_periapsis / Degree);
+    }
+  }
+
+  auto const fill_positions = [this, &ephemeris, &jool_system](
```

Comment:
Please turn this into a bona-fide function in the fixture class.  Too big to lambda.